notebook/tests/selenium/test_display_id.py

```diff
@@ -0,0 +1,160 @@
+"""
+Various output tests.
+"""
+import pytest
+import textwrap
+
+from .utils import wait_for_script_to_return_true
+
+INITIAL_CELLS = [textwrap.dedent("""\
+    ip = get_ipython()
+    from IPython.display import display
+    def display_with_id(obj, display_id, update=False, execute_result=False):
+        iopub = ip.kernel.iopub_socket
+        session = get_ipython().kernel.session
+        data, md = ip.display_formatter.format(obj)
+        transient = {'display_id': display_id}
+        content = {'data': data, 'metadata': md, 'transient': transient}
+        if execute_result:
+            msg_type = 'execute_result'
+            content['execution_count'] = ip.execution_count
+        else:
+            msg_type = 'update_display_data' if update else 'display_data'
+        session.send(iopub, msg_type, content, parent=ip.parent_header)
+    """)]
+
+
+@pytest.fixture
+def get_outputs(notebook):
+    def _get_outputs(cell_idx):
+        return notebook.browser.execute_script(f"""\
+            var cell = Jupyter.notebook.get_cell({cell_idx});
+            return cell.output_area.outputs;
+            """)
+    return _get_outputs
+
+
+@pytest.fixture
+def get_iopub_messages(notebook):
+    def _get_iopub_messages(cell_idx):
+        script = f"return IPython.notebook.get_cell({cell_idx}).iopub_messages"
+        return notebook.browser.execute_script(script)
+    return _get_iopub_messages
+
+
+def test_display_id(prefill_notebook, get_outputs, get_iopub_messages):
+    notebook = prefill_notebook(INITIAL_CELLS)
+    notebook.execute_cell(0)
+
+    #TODO describe test 1
+    notebook.add_and_execute_cell(content=textwrap.dedent("""\
+        display('above')
+        display_with_id(1, 'here')
+        display('below')
+        """))
+    notebook.wait_for_cell_output(1)
+
```

Comment:
Here and at various other points in the test, the JS version calls `wait_for_idle`, which checks if `IPython._status == "idle"`. I tried replicating this in the Python version, but it fails because there's no `_status` on  the IPython object? I'm not sure what's going on here, but the tests don't seem affected by this.